test/CMakeLists.txt

```diff
@@ -212,7 +212,7 @@ endif()
 
 #--stressHistFactory--------------------------------------------------------------------------------
 if(ROOT_roofit_FOUND)
-if(ROOT_xml_FOUND)
+if((xml OR ROOT_xml_FOUND) AND GSL_FOUND) # must use same check as in roofit/CMakeLists.txt
```

Comment:
Why did you need the `xml OR` part? I think this should be `if(ROOT_xml_FOUND AND ROOT_mathmore_FOUND)`.

Reply:
I don't know, I just copied the condition that disables `HistFactory` verbatim

Reply:
It seems `HistFactory` depends on `XMLParser` only, while `hist2workspace` depends on GSL as well. We need to fix this in RooFit. @hageboeck Does `hist2workspace` depend on `mathmore` or just `GSL`, do you know?

Reply:
In summary, the `add_directory(histfactory)` should not be conditional, and the dependencies should be handled in `roofit/histfactory/CMakeLists.txt` for `xml` and `GSL`.